drivers/serial/serial.c

```diff
@@ -1363,12 +1418,18 @@ static ssize_t uart_write(FAR struct file *filep, FAR const char *buffer,
 #endif
 
       flags = enter_critical_section();
-      ret = uart_irqwrite(dev, buffer, buflen);
+      ret = uart_irqwritev(dev, uio);
       leave_critical_section(flags);
 
       return ret;
     }
 
+  buflen = nwritten = uio_resid(uio);
+  if (nwritten < 0)
```

Comment:
why add the additional check

Reply:
because user-given iovecs can contain more than SSIZE_MAX.
the apis like readv can't handle that large requests.